src/index.jsx

```diff
@@ -1,30 +1,70 @@
 import 'babel-polyfill';
-
 import React from 'react';
 import ReactDOM from 'react-dom';
-import { identifyAuthenticatedUser, sendPageEvent } from '@edx/frontend-analytics';
+import { identifyAuthenticatedUser, sendPageEvent, configureAnalytics, initializeSegment } from '@edx/frontend-analytics';
+import LoggingService from '@edx/frontend-logging';
+import { getAuthenticatedAPIClient } from '@edx/frontend-auth';
 
-import './config/analytics';
-import configureStore from './config/configureStore';
-import apiClient from './config/apiClient';
+import { configuration } from './environment';
 import { handleRtl } from './i18n/i18n-loader';
+import configureStore from './store';
+import { configureProfileApiService } from './profile';
+import { configureUserAccountApiService } from './common';
 
 import './index.scss';
-
 import App from './components/App';
```

Comment:
Okay!  So I added stuff here.  This file is ugly and could be factored better, but that fact aside, this is where the magic happens now.  

1. apiClient is now created here, first thing, because nearly everything depends on it.  Rather than it being imported here and there and everywhere, it's passed into code that depends on it via configuration.  
2. After apiClient has determined we're allowed to be here (via `ensurePublicOrAuthenticationAndCookies`), we then configure everything else we need.  The redux store, services, segment, analytics, and RTL.  They all depend on some combination of the apiClient and our configuration, so we hand those two to them here.  Prior to this, they all just imported one or both, which created a hard dependency on those files knowing that there was a file called "apiClient.js" or "environment.js".  

Again, this could be cleaned up and improved, but gathering it all together here means each of these things is no longer reaching out to find the config/apiClient themselves.

Reply:
Not looking at code, but just your comments.  We just need to ensure we don't lose the identify/page event timing that comes after getting user info.